homeassistant/components/mikrotik/hub.py

```diff
@@ -162,15 +166,15 @@ def update_devices(self):
         device_list = {}
         try:
             self.all_devices = self.get_list_from_interface(DHCP)
-            if self.support_wireless:
+            if self.support_wireless or self.support_capsman:
                 _LOGGER.debug("wireless is supported")
                 for interface in [CAPSMAN, WIRELESS]:
                     wireless_devices = self.get_list_from_interface(interface)
                     if wireless_devices:
                         _LOGGER.debug("Scanning wireless devices using %s", interface)
                         break
 
-            if self.support_wireless and not self.force_dhcp:
+            if (self.support_wireless or self.support_capsman) and not self.force_dhcp:
                 device_list = wireless_devices
             else:
                 device_list = self.all_devices
```

Comment:
Now that we are checking for `capsman` support and `wireless` support we need to prioritize `capsman`. i suggest the below code:
```
            if self.support_capsman:
                _LOGGER.debug("Hub is a CAPSman manager")
                device_list = self.get_list_from_interface(CAPSMAN)
            elif self.support_wireless:
                _LOGGER.debug("Hub supports wireless Interface")
                device_list = self.get_list_from_interface(WIRELESS)

            if not device_list or self.force_dhcp:
                device_list = self.all_devices
                _LOGGER.debug("Falling back to DHCP for scanning devices")
            if self.arp_enabled:
                arp_devices = self.get_list_from_interface(ARP)
```

Reply:
I've implemented your suggestion but i also needed to change the wireless params section:
````python
 if self.support_capsman or self.support_wireless:
  # if wireless is supported then wireless_params are params
  self.devices[mac].update(
    wireless_params=device_list[mac], active=True
  )
  continue
````

Reply:
👍 good catch. You need to delete `wireless_devices = {}` as it is no longer used.

Reply:
Oh right. Missed that.
I don't program in python that often.

Reply:
I have been looking at the code again and i found that we do need to keep 2 variables `wireless_devices` and `device_list` because there might be a scenario where we have wireless devices but `force_dhcp` is enabled so `device_list` will not be the same as `wireless_devices`.
For that i suggest the code as follows:
```python
def update_devices(self):
        """Get list of devices with latest status."""
        arp_devices = {}
        device_list = {}
        wireless_devices = {}

        try:
            self.all_devices = self.get_list_from_interface(DHCP)
            if self.support_capsman:
                _LOGGER.debug("Hub is a CAPSman manager")
                device_list = wireless_devices = self.get_list_from_interface(CAPSMAN)
            elif self.support_wireless:
                _LOGGER.debug("wireless is supported")
                device_list = wireless_devices = self.get_list_from_interface(WIRELESS)

            if not device_list or self.force_dhcp:
                device_list = self.all_devices
                _LOGGER.debug("Falling back to DHCP for scanning devices")
```
and you will need to revert the below code as follows:
```python
 if mac in wireless_devices:
  # if wireless or capsman is supported then wireless_params are params
  self.devices[mac].update(
    wireless_params=wireless_devices[mac], active=True
  )
  continue
```

Reply:
done

Reply:
👍